### PR TITLE
allow overriding tsconfig.json compiler options

### PR DESCRIPTION
**Pull request by niieani**
https://github.com/TypeStrong/ts-node/issues/126

### CODE REVIEW

File: src/ts-node.ts [hunk 1]
```diff
@@ -97,7 +98,8 @@ function readConfig (options: Options, cwd: string, ts: TSCommon) {
       inlineSources: true,
       declaration: false,
       noEmit: false
-    }
+    },
+    options.compilerOptions
```
Comment (blakeembrey) [inline src/ts-node.ts]: You can't be allowed to override these options, you should put it between our options and the config options.

**Comment by blakeembrey** [comment]
Do you think we can expose CLI flags for this too? Either as JSON, I guess, or using https://www.npmjs.com/package/subarg.

**Comment by niieani** [comment]
@blakeembrey JSON CLI addition would be good indeed. I take your point regarding not allowing overriding of:
```
    {
      sourceMap: false,
      inlineSourceMap: true,
      inlineSources: true,
      declaration: false,
      noEmit: false
    },
```

I'll have to come back to this at a later time.

**Comment by blakeembrey** [comment]
Closed with https://github.com/TypeStrong/ts-node/pull/128.

**Comment by niieani** [comment]
@blakeembrey awesome, thanks!

**Pull request closed without merging**